machine-learning-image-classification-with-javascript-and-nyckel/index.md

```diff
@@ -0,0 +1,327 @@
+      { 
+        d.preventDefault();
+        d.stopPropagation();
+        });
+
+    // Dragging entry as 'dragenter'
+    
+    $('.upload-area').on('dragenter', 
+    function (d)
+       {
+        d.stopPropagation();
+        d.preventDefault();
+        $("h2").text("Drop");
+    });
+
+    // Dragging over as 'dragover'
+    
+    $('.upload-area').on('dragover', function (d)
+    {
+        d.stopPropagation();
+        d.preventDefault();
+        $("h2").text("Drop");
+    });
+
+    // Drop
+    $('.upload-area').on('drop', 
+    function (d)
+      {
+        d.stopPropagation();
+        d.preventDefault();
+
+        $("h2").text("Checking...");
+
+        var file = d.originalEvent.dataTransfer.files;
+
+        showImage(file[0]);
+
+        resizeAndUploadImage(file[0]);
+    });
+
+    // Open file selector on div click
+    $("#uploadfile").click(function ()
+    {
+        $("#file").click();
+    });
+
+    // file selected
+    $("#file").change(function ()
+    {
+        var file = $('#file')[0].files[0];
+        showImage(file);
+        resizeAndUploadImage(file);
+    });
+});
+function showImage(file)
+{
+    var reader = new FileReader();
+    reader.onload = function (d)
+    {
+        $('.upload-area').css("background-image", "url(" + d.target.result + ")");
+    }
+    reader.readAsDataURL(file);
+}
+
+function resizeAndUploadImage(file)
+{
+    // Ensure it's an image
+    if (file.type.match(/image.*/))
+    {
+        // Load the image
+        var reader = new FileReader();
+        reader.onload = function (readerEvent)
+        {
+            var image = new Image();
+            image.onload = function (imageEvent)
+            {
+                // Resize the image
+                var canvas = document.createElement('canvas'),
+                    max_size = 600,
+                    width = image.width,
+                    height = image.height;
+                if (width > height)
+                {
+                    if (width > max_size)
+                    {
+                        height *= max_size / width;
+                        width = max_size;
+                    }
+                } else
+                {
+                    if (height > max_size)
+                    {
+                        width *= max_size / height;
+                        height = max_size;
+                    }
+                }
+                canvas.width = width;
+                canvas.height = height;
+                canvas.getContext('2d').drawImage(image, 0, 0, width, height);
+                var dataUrl = canvas.toDataURL('image/jpeg');
+                var resizedImage = dataURLToBlob(dataUrl);
+
+                checkImageWithNyckel(resizedImage);
+            }
+            image.src = readerEvent.target.result;
+        }
+        reader.readAsDataURL(file);
+    }
+    else
+    {
+        alert("You must choose an image");
+        resetPage();
+    }
+}
+```

```

Comment:
where is the explanation for all these?

Reply:
Right before the code I have explained what it does...To make it simpler, the function names I have used makes sure that the reader identifies the purpose of each.